pkg/cli/gettoken/gettoken.go

```diff
@@ -0,0 +1,269 @@
+`)
+	getTokenExample = templates.Examples(`
+	# Starts an auth code flow to the issuer url with the client id and the given extra scopes
+	oc get-token --client-id=client-id --issuer-url=test.issuer.url --extra-scopes=email,profile
+
+	# Starts an authe code flow to the issuer url with a different callback address and 5 minute authentication timeout.
+	oc get-token --client-id=client-id --issuer-url=test.issuer.url --callback-addresses=localhost:8343 --authentication-timeout=5m
+`)
+)
+
+const defaultCallbackAddress = "127.0.0.1:8000"
+
+type GetTokenOptions struct {
+	IssuerURL             string
+	ClientID              string
+	ClientSecret          string
+	ExtraScopes           []string
+	CallbackAdresses      []string
+	CACertFilename        string
+	InsecureTLS           bool
+	AuthenticationTimeout time.Duration
+	AutoOpenBrowser       bool
+
+	authenticator oidc.Authenticator
+	tokenCache    *tokencache.Repository
+	credWriter    *credwriter.Writer
+	tokenCacheDir string
+
+	genericiooptions.IOStreams
+}
+
+func NewGetTokenOptions(streams genericiooptions.IOStreams) *GetTokenOptions {
+	return &GetTokenOptions{
+		IOStreams:             streams,
+		AuthenticationTimeout: 3 * time.Minute,
+		CallbackAdresses:      []string{defaultCallbackAddress},
+	}
+}
+
+func NewCmdGetToken(f kcmdutil.Factory, streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewGetTokenOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:     "get-token --oidc-client-id=CLIENT_ID --oidc-issuer-url=ISSUER_URL",
+		Short:   "Experimental: Get token from external OIDC issuer as credentials exec plugin",
+		Long:    getTokenLong,
+		Example: getTokenExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(f, cmd, args))
+			kcmdutil.CheckErr(o.Validate())
+			kcmdutil.CheckErr(o.Run())
+		},
+	}
+
+	cmd.Flags().StringVar(&o.IssuerURL, "issuer-url", o.IssuerURL, "Issuer URL of the external OIDC provider")
+	cmd.Flags().StringVar(&o.ClientID, "client-id", o.ClientID, "Client ID of the user managed by the external OIDC provider")
+	cmd.Flags().StringVar(&o.ClientSecret, "client-secret", o.ClientSecret, "Client Secret of the user managed by the external OIDC provider. Optional.")
+	cmd.Flags().StringSliceVar(&o.ExtraScopes, "extra-scopes", o.ExtraScopes, "Extra scopes for the auth request to the external OIDC provider. Optional.")
+	cmd.Flags().StringSliceVar(&o.CallbackAdresses, "callback-addresses", o.CallbackAdresses, "Callback addresses where external OIDC issuer redirects to after flow is completed. If multiple is passed, command will try to use in an order. Defaults to localhost:8000.")
+	cmd.Flags().DurationVar(&o.AuthenticationTimeout, "authentication-timeout", o.AuthenticationTimeout, "Duration indicates that how long command should wait the authe flow is completed. Defaults to 3 minutes.")
+	cmd.Flags().BoolVar(&o.AutoOpenBrowser, "auto-open-browser", o.AutoOpenBrowser, "Specify browser is automatically opened or not.")
```

Comment:
Automatically opening browser is one of the annoying things not only from me but also several people :). Because my default browser is set to something but I want to open it in different browser by copy-paste. We can have this option similar to all other cred exec plugins. But I'm open to suggestions about what would be the default value (true or false).